sdk/src/react/ui/modals/_internal/components/actionModal/ActionModal.test.tsx

```diff
@@ -0,0 +1,165 @@
+
+		expect(screen.getByTestId('spinner')).toBeInTheDocument();
+		expect(walletResult.current.isLoading).toBe(true);
+	});
+
+	it('Should show error message when useWallet returns isError as true', async () => {
+		// Create a spy on useWallet just for this test
+		const useWalletSpy = vi.spyOn(walletModule, 'useWallet');
+
+		useWalletSpy.mockReturnValue({
+			wallet: null,
+			isLoading: false,
+			isError: true,
+		});
+
+		render(<ActionModal {...defaultProps} />);
+
+		expect(screen.getByTestId('error-loading-text')).toBeInTheDocument();
+		expect(screen.getByText('Error loading modal')).toBeInTheDocument();
+		expect(screen.getByTestId('error-loading-wrapper')).toBeInTheDocument();
+
+		expect(screen.queryByText('Modal Content')).not.toBeInTheDocument();
+		expect(screen.queryByTestId('test-button')).not.toBeInTheDocument();
+
+		// Restore the spy to not affect other tests
+		useWalletSpy.mockRestore();
+	});
+
+	describe('switch chain', async () => {
+		const { result: walletResult } = renderHook(() => walletModule.useWallet());
+
+		expect(walletResult.current.isLoading).toBe(true);
+		render(<ActionModal {...defaultProps} modalLoading={false} />);
+
+		// while wallet is loading, the modal should show the loading spinner
+		expect(screen.getByTestId('spinner')).toBeInTheDocument();
+
+		await waitFor(() => {
+			expect(walletResult.current.isLoading).toBe(false);
+		});
+
+		expect(walletResult.current.isError).toBe(false);
+		expect(walletResult.current.wallet?.address).toBeDefined();
+
+		const walletChainId = await walletResult.current.wallet?.getChainId();
+		// wallet is connected to mainnet (1)
+		expect(walletChainId).toBe(mainnet.id);
+
+		expect(screen.getByText('Modal Content')).toBeInTheDocument();
+		const testButton = screen.getByTestId('test-button');
+
+		it('Should show switch chain modal when chain mismatch (wallet is not a Sequence WaaS or Sequence Ecosystem WaaS)', async () => {
+			fireEvent.click(testButton);
+			switchChainModal$.state.chainIdToSwitchTo.set(polygon.id);
+
+			render(<SwitchChainModal />);
+
+			expect(screen.getByText('Wrong network')).toBeInTheDocument(); // title of the switch chain modal
+		});
+
+		it('Should automatically switch chain without rendering switch chain modal when chain mismatch (wallet is a Sequence WaaS or Sequence Ecosystem WaaS)', async () => {
+			const { result: connectResult } = renderHook(() => useConnect(), {
+				useEmbeddedWallet: true,
+			});
+			// embedded wallet default chain is Polygon 137, so there is a chain mismatch already
+			const { result: walletResult } = renderHook(
+				() => walletModule.useWallet(),
+				{
+					useEmbeddedWallet: true,
+				},
+			);
+
+			await waitFor(() => {
+				connectResult.current.connectAsync({
+					connector: connectResult.current.connectors[0],
+				});
+			});
+
+			render(
+				<ActionModal
+					{...defaultProps}
+					modalLoading={false}
+					_connector={connectResult.current.connectors[0]}
+				/>,
+			);
```

Comment:
Since we want to test this with the embedded wallet to true
```suggestion
			render(
				<ActionModal
					{...defaultProps}
					modalLoading={false}
				/>,
				 {
				useEmbeddedWallet: true
			}
			);
```

Reply:
This saves the day 🌻 It also explains why I get different connectors for testing file and `ActionModal` component 🤌🏼